rlp/sedes/serializable.py

```diff
@@ -174,35 +144,59 @@ def serialize(cls, obj):
             raise ObjectSerializationError(obj=obj, sedes=cls, list_exception=e)
 
     @classmethod
-    def deserialize(cls, serial, mutable=False, **kwargs):
+    def deserialize(cls, serial, **extra_kwargs):
         try:
             values = cls._meta.sedes.deserialize(serial)
         except ListDeserializationError as e:
             raise ObjectDeserializationError(serial=serial, sedes=cls, list_exception=e)
 
-        if mutable:
-            value_kwargs = merge_args_to_kwargs(values, {}, cls._meta.field_names)
-            return cls.create_mutable(**value_kwargs, **kwargs)
-        else:
-            immutable_args = tuple(make_immutable(arg) for arg in values)
-            value_kwargs = merge_args_to_kwargs(immutable_args, {}, cls._meta.field_names)
-            return cls.create_immutable(**value_kwargs, **kwargs)
+        args_as_kwargs = merge_args_to_kwargs(values, {}, cls._meta.field_names)
+        return cls(**args_as_kwargs, **extra_kwargs)
 
+    def copy(self, *args, **kwargs):
+        missing_overrides = set(
+            self._meta.field_names
+        ).difference(
+            kwargs.keys()
+        ).difference(
+            self._meta.field_names[:len(args)]
+        )
+        unchanged_kwargs = {
+            key: copy.deepcopy(value)
+            for key, value
+            in self.as_dict().items()
+            if key in missing_overrides
+        }
+        combined_kwargs = dict(**unchanged_kwargs, **kwargs)
+        all_kwargs = merge_args_to_kwargs(args, combined_kwargs, self._meta.field_names)
+        return type(self)(**all_kwargs)
```

Comment:
This felt like a long way around. Can most of this method roughly compress to:

```py
merged_values = merge(self.as_dict(deepcopy=True), kwargs)
return type(self)(**merged_values)
```

Reply:
If we're merging the kwargs on top of `as_dict()` can't we lazily leave in the parts from `as_dict()` that are also supplied in the `*args, **kwargs`?

```py
all_kwargs = merge_args_to_kwargs(args, kwargs, field_names)
new_fields = merge(self.as_dict(), all_kwargs)
# ^ why do we care if as_dict left in keywords that are supplied in all_kwargs?
```

I imagine that what happened is that I missed some edge case that you thought of, and it should probably be documented.

Reply:
So `args` and `kwargs` are not guaranteed to contain **all** parameters, and the `merge_args_to_kwargs` enforces that all parameters have been supplied (and no extras, and no duplicates).  I didn't want to write a 3rd function for the merger so I did it inline here.  We could extract to a stand-alone but the same logic will still be required.